### PR TITLE
chore: Updates Atlas Go SDK

### PR DESCRIPTION
**Pull request by lantoli**
Automatic update for MongoDB Atlas Go Client SDK

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -24,7 +24,7 @@ require (
 	github.com/spf13/cast v1.5.1
 	github.com/zclconf/go-cty v1.14.1
 	go.mongodb.org/atlas v0.35.0
-	go.mongodb.org/atlas-sdk/v20231001001 v20231001001.1.0
+	go.mongodb.org/atlas-sdk/v20231001002 v20231001002.0.0
```
Comment (andreaangiolillo) [inline go.mod:27]: It does not seems to use SDK preview. 🤔 what am I missing?

Comment (lantoli) [reply go.mod:27]: correct, this is the latest published version

Comment (andreaangiolillo) [reply go.mod:27]: I don't understand why this PR is linked to the PR that talks about SDK preview

**Comment by lantoli** [comment]
Some failing checks but due to unstable test infra, will merge and follow up that tests eventually pass

File: go.mod (hunk 1 above)
Comment (lantoli) [reply go.mod:27]: sorry, the other PR was mainly to fix the SDK Update, And additionally I added the possibility to use the preview sdk locally, never merging it

**Review by andreaangiolillo — CHANGES_REQUESTED** [review]
Tests are failing

File: go.mod (hunk 1 above)
Comment (andreaangiolillo) [reply go.mod:27]: the preview can be used locally without the need to open a PR in terraform

**Pull request closed without merging**